Use Webserver

krishnans2006 opened the pull request:
Compatibility with https://github.com/tj-uav/FlightSoftware/pull/11 to use a new HTTP-based image transfer system, plus more fixes.

This is a preliminary PR before merging into https://github.com/tj-uav/GroundStation/pull/73.

Warning: May cause conflicts with uncommitted changes from @pizzalemon, although I would recommend merging this into `uavhandler` first and then resolving the (hopefully very few) conflicts with that code later.

pizzalemon commented on client/src/App.js:
```diff
@@ -34,9 +34,9 @@ const App = () => {
-					</Route>
+					{/*<Route path="/antenna-tracker">*/}
+					{/*	<AntennaTracker />*/}
+					{/*</Route>*/}
```
Why commit this though.

krishnans2006 replied:
Cause Antenna Tracker is irrelevant

pizzalemon commented on client/src/components/FlightMap.js:
```diff
@@ -195,7 +195,8 @@ const FlightPlanMap = props => {
 					attribution='&amp;copy <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
-					url={online ? "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png" : "/map/{z}/{x}/{y}.png"}
+					// url={online ? "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png" : "/map/{z}/{x}/{y}.png"}
+					url={"https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"}
```
This change shouldn't be commited.

pizzalemon commented on client/src/pages/FlightData/tabs/Logs.js:
```diff
@@ -86,6 +87,7 @@ const Logs = () => {
 	return (
 		<StyledContainer>
 			<CheckboxList name="logFilter" onChange={(e) => {
+					console.log(e.target.value)
```
This shouldn't be commited.

pizzalemon commented on .gitignore:
```diff
@@ -152,6 +152,7 @@ cython_debug/
 !/server/assets/odlc_images/sample.png
 /server/handlers/uav/uav_mission.txt
 /server/handlers/ugv/ugv_mission.txt
+/client/public/map/*/*/*
```
Very relevant to commit.

pizzalemon commented on server/groundstation.py:
```diff
@@ -92,14 +94,26 @@ def ugv_thread(self):
     def image_thread(self):
-        if not self.config["uav"]["images"]["type"] == "prod":  # Initialize a socket connection
-            self.image.socket_connect()
+        if self.config["uav"]["images"]["type"] == "prod":  # Initialize a socket connection
```
Socket connection??? Thought we were getting rid of that.

pizzalemon commented on server/groundstation.py:
```diff
@@ -92,14 +94,26 @@ def ugv_thread(self):
-            self.image.socket_connect()
+        if self.config["uav"]["images"]["type"] == "prod":  # Initialize a socket connection
+            while True:
+                time.sleep(1)
```
Bruh moment.

pizzalemon commented on server/handlers/image.py:
```diff
@@ -67,14 +52,14 @@ def retreive_images(self):
 
     def process_image(self, image):
-        if random() < 0.05:  # 5% chance that the image is of an ODLC
+        if random() < 1:  # 5% chance that the image is of an ODLC
```
Why the change here

pizzalemon commented on server/handlers/image.py:
```diff
@@ -67,14 +52,14 @@ def retreive_images(self):
-                randint(0, 360),
+                self.gs.uav.lat,
+                self.gs.uav.lon,
+                self.gs.uav.orientation["yaw"],
```
Very relevant to Flight Software server. 

You might want to split up this PR into two commits or something.

pizzalemon commented on server/handlers/ugv/dummy.py:
```diff
@@ -69,7 +74,7 @@ def readmission(filename):
     def __init__(self, gs, config):
         self.logger = logging.getLogger("groundstation")
-        self.gs = gs
+        self.gs: GroundStation = gs
```
I don't really care that much, but if you want you could put all these type annotations and small other stuff into a separate code cleanup commit.

pizzalemon commented on server/handlers/ugv/dummy.py:
```diff
@@ -116,8 +121,12 @@ def update(self):
+                self.lon = self.droppos["drop"]["longitude"]# + (random.random() - 0.5) / 2000
+            self.lat += (random.random() - 0.5) / 80000
+            self.lon += (random.random() - 0.5) / 80000
+            # ^^ remove after frr
```
^^ remove after frr

for real

krishnans2006 committed:
Update Image retreival to use new HTTP-based system + more

More: Use better type hints for GroundStation-dependent classes, attempt filter for groundstation vs autopilot logs, other bugfixes

pizzalemon reviewed (approved):
Alright, fine.
